### ISSUE TITLE
Inconsistent character rendering

### ISSUE THREAD

**Issue by inoda** [unidoc/unidoc#378]
We're running into a really funky issue using the Invoice generator.

When calling SetBuyerAddress/SetSellerAddress, and the Name has a hyphen in it (e.g. "John-Doe") sometimes it is rendered as "John-Doe" and sometimes it is rendered as "John›Doe".

This issue is inconsistent. On server restart sometimes the issue is gone. Whether or not the issue exists seems to be the same until server restart.

**Comment by inoda** [comment]
Here is code that can reproduce the issue:

```
	c := creator.New()
	c.NewPage()
	c.Draw(buildInvoicePage(c))
```

```
func buildInvoicePage(c *creator.Creator) *creator.Invoice {
	pdfInvoice := c.NewInvoice()
	pdfInvoice.SetNumber("#123")
	pdfInvoice.AddInfo("Created", "3/22/19")
	pdfInvoice.AddInfo("Amount", "123.12")
	pdfInvoice.SetBuyerAddress(&creator.InvoiceAddress{
		Name:  "John-Doe",
		Email: "example@example.com",
	})

	return pdfInvoice
}
```

This was following the example here: https://unidoc.io/news/simple-invoices

**Comment by gunnsth** [comment]
@inoda The output looks OK on a Windows machine (go 1.11.4). Can you attach one of the broken and one of the OK files?  Comparing those will be useful to get to the root of the problem.

**Comment by inoda** [comment]
[bad.pdf](https://github.com/unidoc/unidoc/files/2999664/bad.pdf)
[ok.pdf](https://github.com/unidoc/unidoc/files/2999665/ok.pdf)

@gunnsth Files are attached above. I am on a mac (10.13.4) and go 1.12. Opening with Acrobat Reader as well.

**Comment by gunnsth** [comment]
I have looked into this and the reason for the differences is that it is mapping '-' to both 0x2D and 0xAD and the ok.pdf case uses 0x2D representation and bad.pdf uses 0xAD.  Both should work (defined in WinAnsiEncoding).

There are two problems here:

- Inconsistency in the output:
The reason for the inconsistency is because the encoding map has two entries for '-'.  We will change it so that it will always pick the lower charcode for a given rune (if non-unique). This will give a consistent output.
- The Encoding needs to be specified as WinAnsiEncoding explicitly.
It appears when this is not defined it is assuming StandardEncoding. The character code 0xAD corresponds to the symbol "›" in the StandardEncoding so this appears to be what is being used here.

We should be able to resolve both of those soon.

**Comment by inoda** [comment]
@gunnsth Thanks for the explanation and quick turnaround! Really appreciate it. Please let me know when this is resolved and what changes we need to make on our end when it is ready. Currently we are using `dep` and are on version `v3.0.0-alpha.2`.

**Comment by gunnsth** [comment]
@inoda I have pushed some code that should fix this issue in branch v3-fix-stdfont-encoding .  Would be great if you can try it out.

**Comment by inoda** [comment]
@gunnsth Looks good on my end!

**Comment by inoda** [comment]
@gunnsth Are these changes going to be merged into a different branch? I assume we won't be pointing to `v3-fix-stdfont-encoding` in production.

**Comment by gunnsth** [comment]
@inoda Expect to go into v3 branch early this week and we'll be tagging a new release soon.

**Comment by gunnsth** [comment]
Merged into v3 in PR #384

**Comment by gunnsth** [comment]
@inoda This is included in new pre-release v3.0.0-alpha.3  that was released today. 
 https://github.com/unidoc/unidoc/releases/tag/v3.0.0-alpha.3

**Comment by inoda** [comment]
@gunnsth Thank you for such a quick fix and good communication!
